mastodon/src/main/java/org/joinmastodon/android/ui/text/ClickableLinksDelegate.java

```diff
@@ -90,21 +99,37 @@ public boolean onTouch(MotionEvent event) {
 			}
 		}
 		if(event.getAction()==MotionEvent.ACTION_UP && selectedSpan!=null){
+			longClickHandler.removeCallbacks(copyTextToClipboard);
 			view.playSoundEffect(SoundEffectConstants.CLICK);
 			selectedSpan.onClick(view.getContext());
-			hlPath=null;
-			selectedSpan=null;
-			view.invalidate();
+			resetAndInvalidate();
 			return false;
 		}
 		if(event.getAction()==MotionEvent.ACTION_CANCEL){
-			hlPath=null;
-			selectedSpan=null;
-			view.invalidate();
+			resetAndInvalidate();
 			return false;
 		}
 		return false;
 	}
+
+	Runnable copyTextToClipboard = () -> {
```

Comment:
Please move the field to the top of the class and make it private (because it isn't used outside of the class).